### PR TITLE
[backport-pr] Select correct remote of the master branch.

### LINKED ISSUE #12314: dev/tools/merge-pr.sh dies silently when the current branch has no remote
The call 
https://github.com/coq/coq/blob/d80458841316ca7edf7317ef412142e27133c931/dev/tools/merge-pr.sh#L84
will cause the script to exit (due to `set -e`) silently if there is no remote.  Perhaps there should be an `|| true` at the end?

### PR DESCRIPTION
**Pull request by Zimmi48**
**Kind:** infrastructure.

### CODE REVIEW

File: dev/tools/backport-pr.sh [hunk 1]
```diff
@@ -30,7 +30,13 @@ while [[ $# -gt 0 ]]; do
     esac
 done
 
-MASTER=origin/master
+REMOTE=$(git config --get "branch.master.remote")
```
Comment (JasonGross) [inline dev/tools/backport-pr.sh:33]: There's the same issue here as I reported in https://github.com/coq/coq/issues/12314.  You'll never actually trigger the `if` branch without making this change
```suggestion
REMOTE=$(git config --get "branch.master.remote" || true)
```

**Review by JasonGross — CHANGES_REQUESTED** [review]

File: dev/tools/backport-pr.sh (hunk 1 above)
Comment (Zimmi48) [reply dev/tools/backport-pr.sh:33]: Thanks!

Comment (Zimmi48) [reply dev/tools/backport-pr.sh]: @JasonGross Fixed in this tool and all the others.

**Review by JasonGross — APPROVED** [review]

**Comment by JasonGross** [comment]
I think you need to put the "fixes #xxxxx" in the body of the commit message rather than the title for GitHub to link the issue to this PR

**Comment by Zimmi48** [comment]
Added the direct link.

**Comment by Zimmi48** [comment]
@JasonGross Do you want to self-assign this PR?

**Pull request merged**